bcipy/signal/tests/model/test_pca_rda_kde.py

```diff
@@ -72,13 +72,11 @@ def setUp(self):
         np.random.seed(0)
 
     def test_pca(self):
-        var_tol = 0.95
-
         # .fit() then .transform() should match .fit_transform()
-        pca = ChannelWisePrincipalComponentAnalysis(num_ch=self.num_channel)
-        pca.fit(self.x, var_tol=var_tol)
+        pca = ChannelWisePrincipalComponentAnalysis(n_components=0.9, num_ch=self.num_channel)
```

Comment:
Is `n_components=0.9` value recommended? If so, can we add a short comment about it?

Reply:
We set this value in a relatively arbitrary way, and further tuning might be good.
I just used the same value in tests as is used in the main code, but this need not be matching. (If we change the value used in tests, we would just need to update the expected test output accordingly).

I'm not sure what info to add in a comment - do you have particular thoughts/questions about it?

Reply:
This value (i.e. variance of the data) is "typically" chosen in the literature to be between 0.95-0.99. I am just trying to find a justification as to why should we set it as 0.9, or 0.95, or 0.97 etc. 

I realized you use the same value in the main code as well. I just thought this choice of parameter could use a source/justifying paper.

Another reason is that I've been specifically asked the number of principal components by a manuscript reviewer before. If we use a specific value like this in our papers the question's very likely to come up I think.

Reply:
Do you have references for the values you cite as typical? 

Our two opposing objectives when setting this value are: 
1. Pick a value low enough to result in compression. In our case, we don't care about storing the transformed data, but rather we care about the resulting speedup of model fitting/predicting for RDA and KDE steps
2. Pick a value high enough to preserve structure in the data.

Given these two conflicting goals, however, the choice of tradeoff between them becomes a bit arbitrary. We picked the value 0.9 because we assumed that this will likely be on the "safe" end of the spectrum, in terms of basically preserving the structure of the data, but still manages to give some decent compression.

In past work, I have seen folks try to automatically determine the number of components by considering whether the residual (`original_signal - rank_K_reconstructed_signal`) "looks indistinguishable from noise" (according to a problem-specific noise model and a carefully chosen statistical test). Notice that this does not solve the problem, but rather trades one arbitrary choice ("what % variance is sufficient for my task?") for another arbitrary choice ("what statistical test, noise model, and p-value are sufficient for my task?").

I think this, or similar adaptive approaches, are probably overkill for the moment.

Would it be useful for me to add this discussion as documentation, just to explain the tradeoff? At the end of the day, someone will still need to make an arbitrary choice. (Even if this makes a reviewer uncomfortable!)